crates/precompile/Cargo.toml

```diff
@@ -14,37 +14,42 @@ exclude = ["build.rs", "src/blob/kzg_settings/*.txt"]
 [dependencies]
 revm-primitives = { path = "../primitives", version = "1.1.2", default-features = false }
 bn = { package = "substrate-bn", version = "0.6", default-features = false }
-k256 = { version = "0.13", default-features = false, features = ["ecdsa"] }
 num = { version = "0.4.0", default-features = false, features = ["alloc"] }
 once_cell = { version = "1.17", default-features = false, features = ["alloc"] }
 ripemd = { version = "0.1", default-features = false }
+sha2 = { version = "0.10", default-features = false }
+
+# Optional KZG precompile
+c-kzg = { git = "https://github.com/ethereum/c-kzg-4844", default-features = false, optional = true }
+
+# Either one enable the ``
+k256 = { version = "0.13", default-features = false, features = [
+    "ecdsa",
+], optional = true }
 secp256k1 = { version = "0.27.0", default-features = false, features = [
     "alloc",
     "recovery",
 ], optional = true }
-sha2 = { version = "0.10.5", default-features = false }
-sha3 = { version = "0.10.7", default-features = false }
-c-kzg = { git = "https://github.com/ethereum/c-kzg-4844", default-features = false, optional = true }
 
 [features]
 default = ["std", "c-kzg", "secp256k1"]
 std = [
     "revm-primitives/std",
-    "k256/std",
     "num/std",
     "once_cell/std",
     "ripemd/std",
     "sha2/std",
-    "sha3/std",
     "c-kzg?/std",
+    "k256?/std",
     "secp256k1?/std",
 ]
 
-# The following features may not work on all platforms as they depend on C libraries.
-# Enables the KZG point evaluation precompile.
+# * Enables the KZG point evaluation precompile.
 c-kzg = ["dep:c-kzg", "revm-primitives/c-kzg"]
 
-# Use `secp256k1` as a faster alternative to `k256`.
-# The problem that `secp256k1` has is it fails to build for `wasm` target on Windows and Mac as it is c lib.
-# In Linux it passes. If you don't require to build wasm on win/mac, it is safe to use it and it is enabled by default.
+# Either enables the ecRecover precompile
+k256 = ["dep:k256"]
```

Comment:
Better not, it is fine having k256 as default, it can be a footgun if not included and k256 build everywhere. Only downside is k256 is always included.

Reply:
makes sense, unfortunate that you can't disable it when `secp256k1` is enabled because Cargo doesn't allow this